server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/routes/CassandraMigrationRoutes.java

```diff
@@ -72,15 +72,28 @@ public void define(Service service) {
                 response.status(NO_CONTENT);
             } catch (NullPointerException | IllegalArgumentException e) {
                 LOGGER.info("Invalid request for version upgrade");
-                response.status(INVALID_VERSION);
+                throw ErrorResponder.builder()
+                    .statusCode(HttpStatus.BAD_REQUEST_400)
+                    .type(ErrorType.INVALID_ARGUMENT)
+                    .message("Invalid request for version upgrade")
+                    .cause(e.getMessage())
+                    .haltError();
             } catch (IllegalStateException e) {
                 LOGGER.info("The migration requested can not be performed.", e);
-                response.status(MIGRATION_CAN_NOT_BE_PERFORMED);
-                response.body(e.getMessage());
+                throw ErrorResponder.builder()
+                    .statusCode(HttpStatus.GONE_410)
+                    .type(ErrorType.WRONG_STATE)
+                    .message("The migration requested can not be performed")
+                    .cause(e.getMessage())
+                    .haltError();
             } catch (MigrationException e) {
                 LOGGER.error("An error lead to partial migration process", e);
-                response.status(INTERNAL_ERROR);
-                response.body(e.getMessage());
+                throw ErrorResponder.builder()
+                    .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                    .type(ErrorType.SERVER_ERROR)
+                    .message("An error lead to partial migration process")
+                    .cause(e.getMessage())
```

Comment:
We should pass: 

```
    .cause(e)
```

This is more flexible and might enable having stacktrace....

Reply:
Do we really want to show anyone the all stacktrace ?